lib/connectors/memory.js

```diff
@@ -237,6 +244,38 @@ Memory.prototype.updateOrCreate = function (model, data, options, callback) {
   });
 };
 
+Memory.prototype.findOrCreate = function(model, filter, data, callback) {
+  var self = this;
+  var nodes = self._findAll(model, filter);
+
+  if (nodes.length > 0) {
+    if (filter && filter.include) {
+      self._models[model].model.include(nodes[0], filter.include, {}, function(err, nodes) {
```

Comment:
It seems Mongo is using `filter.include` for  Please see: https://github.com/strongloop/loopback-connector-mongodb/blob/master/lib/mongodb.js#L1134-L1137